float64 x
float64 y
float64 z
int64 status
int64 face
